src/cli-constant.js

```diff
@@ -67,58 +277,45 @@ const options = {
-                           cli-override  | default config => config file => CLI options
-                           file-override | default config => CLI options => config file
-                           prefer-file   | default config => config file (if config file is found) or
-                                           default config => CLI options (if no config file is found)
-                           Defaults to cli-override
-  --no-config              Do not look for a configuration file.
-  --find-config-path <path>
-                           Finds and prints the path to a configuration file for a given input file.
-  --ignore-path <path>     Path to a file containing patterns that describe files to ignore.
-                           Defaults to ./.prettierignore.
-  --stdin                  Read input from stdin.
-  --stdin-filepath         Path to the file used to read from stdin.
-  --print-width <int>      Specify the length of line that the printer will wrap on. Defaults to 80.
-  --tab-width <int>        Specify the number of spaces per indentation-level. Defaults to 2.
-  --use-tabs               Indent lines with tabs instead of spaces.
-  --no-semi                Do not print semicolons, except at the beginning of lines which may need them.
-  --single-quote           Use single quotes instead of double quotes.
-  --no-bracket-spacing     Do not print spaces between brackets.
-  --jsx-bracket-same-line  Put > on the last line instead of at a new line.
-  --trailing-comma <none|es5|all>
-                           Print trailing commas wherever possible when multi-line. Defaults to none.
-  --parser <flow|babylon|typescript|postcss|json|graphql>
-                           Specify which parse to use. Defaults to babylon.
-  --cursor-offset <int>    Print (to stderr) where a cursor at the given position would move to after formatting.
-                           This option cannot be used with --range-start and --range-end
-  --range-start <int>      Format code starting at a given character offset.
-                           The range will extend backwards to the start of the first line containing the selected statement.
-                           This option cannot be used with --cursor-offset.
-                           Defaults to 0.
-  --range-end <int>        Format code ending at a given character offset (exclusive).
-                           The range will extend forwards to the end of the selected statement.
-                           This option cannot be used with --cursor-offset.
-                           Defaults to Infinity.
-  --no-color               Do not colorize error messages.
-  --with-node-modules      Process files inside 'node_modules' directory.
-  --version or -v          Print Prettier version.
+  normalized.forEach(normalizedOption => {
+    normalized[normalizedOption.name] = normalizedOption;
+  });
```

Comment:
What are these three lines for?

Reply:
It's a convenient way to use something like `detailOptions["parser"]` in `cli-util.js`

Reply:
This feels a bit too magic to me. Can we make a function instead? Something like `getOption(detailOptions, "parser")`

Reply:
Maybe `detailOptionMap` would be better? since it's in `cli-constant.js`.

Reply:
I guess that’s fine too 👍